client/components/phone-input/test/phone-number__format-number.test.js

```diff
@@ -0,0 +1,220 @@
+
+	test.each( [
+		{ number: '+1', country: 'US', expected: '+1' },
+		{ number: '+14', country: 'US', expected: '+14' },
+		{ number: '+142', country: 'US', expected: '+1 42' },
+		{ number: '+1425', country: 'US', expected: '+1 425' },
+		{ number: '+14256', country: 'US', expected: '+1 425-6' },
+		{ number: '+142565', country: 'US', expected: '+1 425-65' },
+		{ number: '+1425655', country: 'US', expected: '+1 425-655' },
+		{ number: '+14256559', country: 'US', expected: '+1 425-655-9' },
+		{ number: '+142565599', country: 'US', expected: '+1 425-655-99' },
+		{ number: '+1425655999', country: 'US', expected: '+1 425-655-999' },
+		{ number: '+14256559999', country: 'US', expected: '+1 425-655-9999' },
+		{ number: '+1', country: 'CA', expected: '+1' },
+		{ number: '+16', country: 'CA', expected: '+16' },
+		{ number: '+160', country: 'CA', expected: '+1 60' },
+		{ number: '+1604', country: 'CA', expected: '+1 604' },
+		{ number: '+16046', country: 'CA', expected: '+1 604-6' },
+		{ number: '+160465', country: 'CA', expected: '+1 604-65' },
+		{ number: '+1604655', country: 'CA', expected: '+1 604-655' },
+		{ number: '+16046559', country: 'CA', expected: '+1 604-655-9' },
+		{ number: '+160465599', country: 'CA', expected: '+1 604-655-99' },
+		{ number: '+1604655999', country: 'CA', expected: '+1 604-655-999' },
+		{ number: '+16046559999', country: 'CA', expected: '+1 604-655-9999' },
+		{ number: '+90', country: 'TR', expected: '+90' },
+		{ number: '+905', country: 'TR', expected: '+90 5' },
+		{ number: '+9055', country: 'TR', expected: '+90 55' },
+		{ number: '+90555', country: 'TR', expected: '+90 555' },
+		{ number: '+905558', country: 'TR', expected: '+90 555 8' },
+		{ number: '+9055588', country: 'TR', expected: '+90 555 88' },
+		{ number: '+90555888', country: 'TR', expected: '+90 555 888' },
+		{ number: '+905558889', country: 'TR', expected: '+90 555 888 9' },
+		{ number: '+9055588899', country: 'TR', expected: '+90 555 888 99' },
+		{ number: '+90555888999', country: 'TR', expected: '+90 555 888 99 9' },
+		{ number: '+905558889999', country: 'TR', expected: '+90 555 888 99 99' },
+	] )( `Number is formatted as it is typed ($number)`, function ( { number, country, expected } ) {
+		const result = formatNumber( number, countries[ country ] );
+		expect( result ).toEqual( expected );
+	} );
+} );
+
+describe( 'National Format', () => {
+	test.each( [
+		{ number: '4252222222', country: 'US', expected: '(425) 222-2222' },
+		{ number: '05325555555', country: 'TR', expected: '0532 555 55 55' },
+		{ number: '0215369851', country: 'AU', expected: '02 1536 9851' },
+		{ number: '3911711711', country: 'IT', expected: '391 171 1711' },
+		{ number: '5512345678', country: 'BR', expected: '055 1234-5678' },
+		{ number: '9100123456', country: 'IN', expected: '091001 23456' },
+		{ number: '00000123456', country: 'UK', expected: '01234 56' },
+	] )( `Format full length numbers`, function ( { number, country, expected } ) {
```

Comment:
Do we want to add some part of the input to the test name here so the tests have different names? Right now they all have the same name, which might make debugging a single failure a pinch slower.

Reply:
Addressed in e10751336012e40c886f517b08cc4f2cbe15d2e0.